server/src/main/java/io/druid/segment/realtime/plumber/Sink.java

```diff
@@ -204,6 +207,15 @@ private FireHydrant makeNewCurrIndex(long minTimestamp, DataSchema schema)
       if (numHydrants > 0) {
         FireHydrant lastHydrant = hydrants.get(numHydrants - 1);
         newCount = lastHydrant.getCount() + 1;
+        if(!indexSchema.getDimensionsSpec().hasCustomDimensions()) {
+          if (lastHydrant.hasSwapped()) {
```

Comment:
@gianm lastHydrant.hasSwapped() is true when makeNewCurrentIndex is called from the bootstrapSinksFromDisk() path, when the hydrants are created using QueryableIndexSegment